### PR TITLE
fix: edge case

### LINKED ISSUE #99: edge case bug - contract account with low balance
If the staking pool is created on an account with balance = 1_000_000_000_000, then total_stake_shares is 0, and several view functions start to panic

**Note:** I'm preparing a PR to handle this edge case

To reproduce, add this unit test: (the 4th parameter is the account initial balance)

```
    #[test]
    fn test_low_staked_balance() {
        let mut emulator = Emulator::new(
            owner(),
            "KuTCtARNzxZQ3YvXDeLjx83FDqxv2SdQTSbiq876zR7".to_string(),
            zero_fee(),
            STAKE_SHARE_PRICE_GUARANTEE_FUND, //account initial balance, minimum, edge case
        );
        //bob deposits 1M
        let deposit_amount = ntoy(1_000_000);
        emulator.update_context(bob(), deposit_amount);
        emulator.contract.deposit();
        emulator.amount += deposit_amount;

        //bob wants to know how much unstaked balance he has
        emulator.update_context(bob(), 0);
        assert_eq!(
            emulator.contract.get_account_unstaked_balance(bob()).0,
            deposit_amount
        );
        //panic!
        //thread 'tests::test_low_staked_balance' panicked at 'The total number of stake shares can't be 0', src/internal.rs:297:9

    }

```

### PR DESCRIPTION
**Pull request by luciotato**
fixes #99

### CODE REVIEW

File: staking-pool/src/lib.rs [hunk 1]
```diff
@@ -514,10 +518,11 @@ mod tests {
             owner: String,
             stake_public_key: String,
             reward_fee_fraction: RewardFeeFraction,
+            initial_balance: Balance,
```
Comment (luciotato) [inline staking-pool/src/lib.rs:521]: add a 4th parameter for testing, initial balance

File: staking-pool/src/lib.rs [hunk 2]
```diff
@@ -183,6 +183,10 @@ impl StakingContract {
             "The owner account ID is invalid"
         );
         let account_balance = env::account_balance();
+        assert!(
```
Comment (luciotato) [inline staking-pool/src/lib.rs:186]: avoid `total_staked_balance`=0

Comment (evgenykuzyakov) [reply staking-pool/src/lib.rs:186]: Realistically, this is impossible, because the storage required for the contract itself covers this.

File: staking-pool/src/lib.rs [hunk 3]
```diff
@@ -183,6 +183,10 @@ impl StakingContract {
...
+            account_balance > STAKE_SHARE_PRICE_GUARANTEE_FUND,
```
Comment (evgenykuzyakov) [inline staking-pool/src/lib.rs:187]: Why this is strictly higher?

File: staking-pool/src/lib.rs (hunk 2 above)
Comment (luciotato) [reply staking-pool/src/lib.rs:186]: I see. The contract will stop working way before that.
Question: Who pays the storage fees? It comes from the staked NEARs?

File: staking-pool/src/lib.rs (hunk 3 above)
Comment (luciotato) [reply staking-pool/src/lib.rs:187]: because if `account_balance == STAKE_SHARE_PRICE_GUARANTEE_FUND` then in the next line total_staked_balance becomes zero. `total_staked_balance` is used in several share calculations as denominator, so `total_staked_balance==0` leads to div by zero panics. (i.e. `fn num_shares_from_staked_amount_rounded_down/up`)
